File: src/main/java/Electro1D/Protein.java
package main.java.Electro1D;

import java.awt.Color;
import java.awt.Graphics;

/**
 * @author Bader Alharbi
 */
public class Protein {

	int concentration;
	double startY;
	public float scaleFactor;
	public double relativeMigration;
	public double speed;
	public int mw;
	public float charge;
	public String name;
	public String fullName;
	public String abbr;
	public boolean selected;
	public Color color;
	public double x1;
	public double y1;
	public int width;
	public int height;
	public double y1float;
	public int maxPosition;
	public double distance;
	private int decider;
	private int counter;
	public int plotYPos;
	public int plotXPos;

	public void setStartPosition(int x, int y) {
		x1 = x;
		y1 = y;
		startY = y1;
		y1float = y1;
	}

	public void resetDecider() {
		decider = 1;
		counter = 1;
	}

	private void incrPosition() {
		if (counter == decider) {
			y1float = y1float + speed;
			counter = 1;
		} else {
			counter++;
		}
		y1 = (int) y1float;
	}

	public boolean matchPosition(int x, int y) {
		double x2 = x1 + width;
		double y2 = y1 + height;
		int range = 2;
		return (x >= x1 - range && x <= x2 + range && y >= y1 - range && y <= y2 + range);
	}

	public void setWidth(int w) {
		width = w;
	}

	public int getDecider() {
		return decider;
	}

	public void setDecider(int i) {
		decider = i;
	}

	void setHostScaleFactor(float f) {
		scaleFactor = f;
	}

	public void setMaxPosition(int i) {
		maxPosition = i;
	}

	public void setConcentration(int i) {
		concentration = i;
	}

	boolean matchPlotPosition(int x, int y) {
		int range = 3;
		return (x >= plotXPos - range && x <= plotXPos + range && y >= plotYPos - range && y <= plotYPos + range);
	}

	public double getDistance() {
		distance = scaleFactor * (y1 - startY);
		return distance;
	}

	Protein() {
		speed = 0.01D;
		name = "notSet";
		fullName = "notSet";
		abbr = "notSet";
		color = Color.blue;
		height = 2;
		decider = 1;
		counter = 1;
		concentration = 1;
	}

	public Protein(String name, String fullName, String abbr, int i, Color color1) {
		speed = 0.01D;
		color = Color.blue;
		height = 2;
		decider = 1;
		counter = 1;
		this.name = name;
		this.fullName = fullName;
		this.abbr = abbr;
		mw = i;
		color = color1;
		concentration = 1;
	}

	public boolean drawProtein(Graphics g) {
		if (y1 >= maxPosition) {
			return false;
		}
		g.setColor(color);
		if (concentration == 1) {
			g.fillRect((int) x1, (int) y1, width, height);
		} else {
			g.fillRect((int) x1 - concentration, (int) y1, width + concentration * 2, height + concentration);
		}
		incrPosition();
		return true;
	}

	public String toString() {
		return "Protein: name=" + name + " fullName=" + fullName + " MW=" + mw + " rm=" + relativeMigration + " d=" + distance;
	}
}

File: site/swingjs/j2s/main/java/Electro1D/Protein.js
(function(){var P$=Clazz.newPackage("main.java.Electro1D"),p$1={},I$=[[0,'java.awt.Color']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "Protein");

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

C$.$fields$=[['Z',['selected'],'D',['startY','relativeMigration','speed','x1','y1','y1float','distance'],'F',['scaleFactor','charge'],'I',['concentration','mw','width','height','maxPosition','decider','counter','plotYPos','plotXPos'],'S',['name','fullName','abbr'],'O',['color','java.awt.Color']]]

Clazz.newMeth(C$, 'setStartPosition$I$I',  function (x, y) {
this.x1=x;
this.y1=y;
this.startY=this.y1;
this.y1float=this.y1;
});

Clazz.newMeth(C$, 'resetDecider$',  function () {
this.decider=1;
this.counter=1;
});

Clazz.newMeth(C$, 'incrPosition',  function () {
if (this.counter == this.decider) {
this.y1float=this.y1float + this.speed;
this.counter=1;
} else {
++this.counter;
}this.y1=(this.y1float|0);
}, p$1);

Clazz.newMeth(C$, 'matchPosition$I$I',  function (x, y) {
var x2=this.x1 + this.width;
var y2=this.y1 + this.height;
var range=2;
return (x >= this.x1 - range  && x <= x2 + range   && y >= this.y1 - range   && y <= y2 + range  );
});

Clazz.newMeth(C$, 'setWidth$I',  function (w) {
this.width=w;
});

Clazz.newMeth(C$, 'getDecider$',  function () {
return this.decider;
});

Clazz.newMeth(C$, 'setDecider$I',  function (i) {
this.decider=i;
});

Clazz.newMeth(C$, 'setHostScaleFactor$F',  function (f) {
this.scaleFactor=f;
});

Clazz.newMeth(C$, 'setMaxPosition$I',  function (i) {
this.maxPosition=i;
});

Clazz.newMeth(C$, 'setConcentration$I',  function (i) {
this.concentration=i;
});

Clazz.newMeth(C$, 'matchPlotPosition$I$I',  function (x, y) {
var range=3;
return (x >= this.plotXPos - range && x <= this.plotXPos + range  && y >= this.plotYPos - range  && y <= this.plotYPos + range );
});

Clazz.newMeth(C$, 'getDistance$',  function () {
this.distance=this.scaleFactor * (this.y1 - this.startY);
return this.distance;
});

Clazz.newMeth(C$, 'c$',  function () {
;C$.$init$.apply(this);
this.speed=0.01;
this.name="notSet";
this.fullName="notSet";
this.abbr="notSet";
this.color=$I$(1).blue;
this.height=2;
this.decider=1;
this.counter=1;
this.concentration=1;
}, 1);

Clazz.newMeth(C$, 'c$$S$S$S$I$java_awt_Color',  function (name, fullName, abbr, i, color1) {
;C$.$init$.apply(this);
this.speed=0.01;
this.color=$I$(1).blue;
this.height=2;
this.decider=1;
this.counter=1;
this.name=name;
this.fullName=fullName;
this.abbr=abbr;
this.mw=i;
this.color=color1;
this.concentration=1;
}, 1);

Clazz.newMeth(C$, 'drawProtein$java_awt_Graphics',  function (g) {
if (this.y1 >= this.maxPosition ) {
return false;
}g.setColor$java_awt_Color(this.color);
if (this.concentration == 1) {
g.fillRect$I$I$I$I((this.x1|0), (this.y1|0), this.width, this.height);
} else {
g.fillRect$I$I$I$I((this.x1|0) - this.concentration, (this.y1|0), this.width + this.concentration * 2, this.height + this.concentration);
}p$1.incrPosition.apply(this, []);
return true;
});

Clazz.newMeth(C$, 'toString',  function () {
return "Protein: name=" + this.name + " fullName=" + this.fullName + " MW=" + this.mw + " rm=" + new Double(this.relativeMigration).toString() + " d=" + new Double(this.distance).toString() ;
});
})();
;Clazz.setTVer('3.3.1-v4');//Created 2022-04-21 16:53:53 Java2ScriptVisitor version 3.3.1-v4 net.sf.j2s.core.jar version 3.3.1-v4
